projects/js-packages/publicize-components/src/components/connection-icon/index.jsx

```diff
@@ -1,15 +1,23 @@
 import { SocialServiceIcon } from '@automattic/jetpack-components';
+import { useCallback, useState } from '@wordpress/element';
 import PropTypes from 'prop-types';
 
 import './style.scss';
 
 const ConnectionIcon = props => {
 	const { id, serviceName, label, profilePicture } = props;
+	const [ isPictureLoaded, setIsPictureLoaded ] = useState( false );
+	const [ displayPicture, setDisplayPicture ] = useState( !! profilePicture );
```

Comment:
Could we combine these to have one `hasDisplayPicture` flag, that starts as `false` and is updated to `true` by `onLoad`? That could also remove the need for the `onError` handler.

Reply:
That's what I did initially, but the problem is that starting as `false` means that the profile picture doesn't get rendered, which prevents the `onLoad` from firing. There's other ways of hiding the image of course, but adding an extra variable seemed like a good tradeoff in this case.

Reply:
You know what, that sounds like it might work! I'll give it a shot.

Reply:
Fixed in [b19082e](https://github.com/Automattic/jetpack/pull/24736/commits/b19082e17d8b22920902a0ae7dc9efe2c697f012). Thanks!